src/main/java/org/folio/edge/core/utils/test/TestUtils.java

```diff
@@ -1,60 +1,132 @@
+      }
+      if (maxTries == 0 || isLocalPortFree(port)) {
+        return port;
+      }
+      --maxTries;
+    }
   }
 
-  public static void assertLogMessage(Logger logger, int minTimes, int maxTimes, Level logLevel, String expectedMsg,
-      Throwable t, Runnable func) {
-    Appender appender = mock(Appender.class);
+  static void getPortReset() {
+    cyclicPort.set(PORT_MAX);
+  }
+
+  /**
+   * Check a local TCP port.
+   * @param port  the TCP port number, must be from 1 ... 65535
+   * @return true if the port is free (unused), false if the port is already in use
+   */
+  static boolean isLocalPortFree(int port) {
+      try {
+          new ServerSocket(port).close();
+          return true;
+      } catch (IOException e) {
+          return false;
+      }
+  }
 
-    assertNotNull(logger);
-    assertNotNull(func);
+  /**
+   * Unit test for logged content.
+   *
+   * @param logger that is used in test code
+   * @param minTimes minimum number of log events
+   * @param maxTimes maximum number of log events
+   * @param logLevel level to test against (not working, so not checked)
+   * @param expectedMsg expected message for first message
+   * @param t expected Throwable (not working, pass null always)
+   * @param func function to execute which presumably logs
+   */
+  public static void assertLogMessage(Logger logger, int minTimes, int maxTimes, Level logLevel,
+                                      String expectedMsg, Throwable t, Runnable func) {
+    assertLogMessage((org.apache.logging.log4j.core.Logger) logger, minTimes, maxTimes, logLevel, expectedMsg, t, func);
+  }
 
-    try {
-      logger.addAppender(appender);
-      ArgumentCaptor<LoggingEvent> argument = ArgumentCaptor.forClass(LoggingEvent.class);
+  private static void assertLogMessage(org.apache.logging.log4j.core.Logger logger, int minTimes, int maxTimes, Level logLevel,
+                                       String expectedMsg, Throwable t, Runnable func) {
 
-      func.run();
+    AppenderCapture appender = new AppenderCapture();
 
-      verify(appender, atLeast(minTimes)).doAppend(argument.capture());
-      verify(appender, atMost(maxTimes)).doAppend(argument.capture());
+    logger.addAppender(appender);
+    func.run();
+    appender.stop();
+    logger.removeAppender(appender);
 
-      if (logLevel != null)
-        assertEquals(logLevel, argument.getValue().getLevel());
+    Assert.assertTrue(appender.events.size() >= minTimes);
+    Assert.assertTrue(appender.events.size() <= maxTimes);
 
-      if (expectedMsg != null)
-        assertEquals(expectedMsg, argument.getValue().getMessage());
+    Assert.assertNull(t);
 
-      if (t != null) {
-        assertNotNull(argument.getValue().getThrowableInformation());
-        assertEquals(t, argument.getValue().getThrowableInformation().getThrowable());
-      }
-    } finally {
-      logger.removeAppender(appender);
+    if (logLevel != null) {
+      Assert.assertTrue(!appender.events.isEmpty());
```

Comment:
Why do we check this here?  What if we only wanted to check that the log level was set w/o ever having logged a message?

Reply:
If no logging is made, it's not possible to check for log level fired off. But I'm happy to say, that case is true (no failures).

Reply:
right, got it.